dsms/knowledge/utils.py

```diff
@@ -279,16 +282,14 @@ def _commit_updated(buffer: "Dict[str, KItem]") -> None:
     """Commit the buffer for the `updated` buffers"""
     for kitem in buffer.values():
         if _kitem_exists(kitem):
-            _update_attachments(kitem)
             if isinstance(kitem.hdf5, pd.DataFrame):
                 _update_hdf5(kitem.id, kitem.hdf5)
             elif isinstance(kitem.hdf5, type(None)) and _inspect_hdf5(
                 kitem.id
             ):
                 _delete_hdf5(kitem.id)
             _update_kitem(kitem)
-            for key, value in _get_kitem(kitem.id).__dict__.items():
-                setattr(kitem, key, value)
+            _update_attachments(kitem)
```

Comment:
Why ` _update_attachments` is not being called from within the `_update_kitem` function?

Reply:
I think I did this because the attachments and the kitem-fields are independent from each other. Same holds for the hdf5 from the `data_api`. In this manner, it is less nested and the attributes are treated separately from the hdf5 and the attachments.